src/riscv.c

```diff
@@ -105,11 +106,57 @@ riscv_word_t rv_get_reg(riscv_t *rv, uint32_t reg)
     return ~0U;
 }
 
-riscv_t *rv_create(const riscv_io_t *io,
-                   riscv_user_t userdata,
-                   int argc,
-                   char **args,
-                   bool output_exit_code)
+/*
+ * Register non-common stdio
+ *
+ * @rv: riscv
+ * @stdio: 0 or 1 or 2
+ * @new_fptr: FILE * for new redirection
+ *
+ */
+void rv_register_stdio(riscv_t *rv, int stdio_fd, FILE *new_fptr)
```

Comment:
Quote [Wikipedia](https://en.wikipedia.org/wiki/Standard_streams):
> In computer programming, standard streams are preconnected input and output communication channels between a computer program and its environment when it begins execution. The three input/output (I/O) connections are called standard input (stdin), standard output (stdout) and standard error (stderr).

In the proposed changed code, the use of stdio is confusing.

Reply:
I think `rv_remap_stream(riscv_t *rv, int fd, FILE *new_fptr)` is more comprehensive. More comments can be added to classify the parameters.

Reply:
How about `rv_remap_stdstream(riscv_t *rv, int fd, FILE *filev)`?
Here `filev` means a vector of `FILE *`.

Reply:
A fd is given in a call, is it necessary required a vector of `FILE *`?

Reply:
> A fd is given in a call, is it necessary required a vector of `FILE *`?

No, my intention was to encapsulate the pair of a file descriptor and standard streams. This was meant to form a vector, but I misplaced the parameters.

Reply:
> How about rv_remap_stdstream(riscv_t *rv, int fd, FILE *filev)?

Much more sense than `rv_register_stdio`.

> > A fd is given in a call, is it necessary required a vector of `FILE *`?
> 
> No, my intention was to encapsulate the pair of a file descriptor and standard streams. This was meant to form a vector, but I misplaced the parameters.

Got it. So, the signature becomes `rv_remap_stdstream(riscv_t *rv, int *fds, FILE *filev)`. We could assume the length of the vector is 3, otherwise undefined behavior. If any `fd` in `fds` is -1, it fallback to use default stdio.

Reply:
> So, the signature becomes `rv_remap_stdstream(riscv_t *rv, int *fdv, FILE *filev)`. We could assume the length of the vector is 3, otherwise undefined behavior. If any `fd` in `fdv` is -1, it fallback to use default stdio.

To mitigate the risk of passing null or malformed pointers, we could introduce a structure, `fd_stream_pair`, to encapsulate file descriptors and `FILE` pointers. This structure can then be reliably passed as a pointer to the `rv_remap_stdstream` function, ensuring safer and more organized handling of these elements.